extensions/eclipse-che-theia-preferences-provider-extension/src/browser/async-preference-provider.ts

```diff
@@ -0,0 +1,100 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
```

Comment:
Does it mean that `AsyncPreferenceProvider` things were copied from upstream?
Should we have a CQ then?

Reply:
No, for two reasons:
1. I just copied the header, the rest is at most "inspiried by"
2. We're already shipping all of theia with our product, so we already have approval to use that code.